public/src/components/epicTests/articlesViewedEditor.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import {
+  createStyles, FormControl,
+  FormControlLabel,
+  InputLabel,
+  Switch,
+  Theme, Typography,
+  withStyles,
+  WithStyles
+} from "@material-ui/core";
+import EditableTextField from "../helpers/editableTextField"
+import {onFieldValidationChange, ValidationStatus} from "../helpers/validation";
+import {ArticlesViewedSettings, EpicTest} from "./epicTestsForm";
+
+const isNumber = (value: string): boolean => !Number.isNaN(Number(value));
+
+export const defaultArticlesViewedSettings: ArticlesViewedSettings = {
+  maxViews: null,
+  minViews: 5,
+  periodInWeeks: 8,
+};
+
+const styles = ({ spacing, typography}: Theme) => createStyles({
+  button: {
+    marginTop: '20px'
+  },
+  formControl: {
+    marginTop: spacing(2),
+    marginBottom: spacing(1),
+    display: "block",
+  },
+  selectLabel: {
+    fontSize: typography.pxToRem(22),
+    fontWeight: typography.fontWeightMedium,
+    color: "black"
+  },
+  radio: {
+    paddingTop: "20px",
+    marginBottom: "10px"
+  },
+  articlesViewsContainer: {
+    marginTop: '10px',
+    marginLeft: '20px'
+  }
+});
+
+type ArticleViewsFieldName = 'minViews' | 'maxViews' | 'periodInWeeks';
+
+interface Props extends WithStyles<typeof styles> {
+  articlesViewedSettings?: ArticlesViewedSettings,
+  editMode: boolean,
+  onChange: (articlesViewedSettings?: ArticlesViewedSettings) => void,
+  onValidationChange: (isValid: boolean) => void
+}
+
+interface State {
+  validationStatus: ValidationStatus
+}
+
+class ArticlesViewedEditor extends React.Component<Props, State> {
+  state: State = {
+    validationStatus: {}
+  };
+
+  buildField = (fieldName: ArticleViewsFieldName, label: string, articlesViewedSettings: ArticlesViewedSettings) => {
+    const setting: number | null = articlesViewedSettings[fieldName];
+
+    return (<EditableTextField
+      text={ setting ? setting.toString() : ''}
+      onSubmit={ (value: string) => {
+        if (isNumber(value)) {
+          this.props.onChange(
+            {
+              ...articlesViewedSettings || defaultArticlesViewedSettings,
+              [fieldName]: Number(value)
+            }
+          )
+        }
+      }}
+      label={label}
+      helperText="Must be a number"
+      editEnabled={this.props.editMode}
+      validation={
+        {
+          getError: (value: string) => isNumber(value) ? null : 'Must be a number',
+          onChange: onFieldValidationChange(this)(fieldName)
+        }
+      }
+      isNumberField
+    />)
+  };
+
+  onArticlesViewedEditorSwitchChange = (enabled: boolean) => {
+    if (enabled) {
+      this.props.onChange(defaultArticlesViewedSettings);
+    } else {
+      this.props.onChange(undefined);
+    }
+  }
```

Comment:
Do we need the else here? Could it be `enabled && this.props.onChange(defaultArticlesViewedSettings)`?

Reply:
we need to set them specifically as undefined in the case the switch is disabled - once they are undefined, they will be hidden